onnxruntime/core/graph/graph.cc

```diff
@@ -4128,32 +4131,56 @@ Graph::~Graph() {
 }
 
 #if !defined(ORT_MINIMAL_BUILD)
+std::ostream& operator<<(std::ostream& out, const NodeArg& node_arg) {
+  out << "\"" << node_arg.Name() << "\"";
+  if (node_arg.Type()) {
+    out << ": " << *node_arg.Type();
+  } 
+  return out;
+}
+
+std::ostream& operator<<(std::ostream& out, const Node& node) {
+  out << "(\"" << node.Name() << "\"";
+  out << ", ";
+  out << node.OpType();
+  out << ", ";
+  // Use quote so default ONNX domain is shown as ""
+  // rather than misleading empty string.
+  out << "\"" << node.Domain() << "\"";
+  out << ", ";
+  out << node.SinceVersion();
+  out << ") : (";
+  for (auto* x : node.InputDefs()) {
+    if (x->Exists()) {
+      out << *x << ",";
+    } else {
+      out << "" << ",";
```

Comment:
>"" [](http://example.com/codeflow?start=13&length=2)

`out << ""` won't print anything right? also shows up below

Reply:
It's intended. If we create an non-existing NodeArg, I prefer to let user see it. Otherwise, user can't visually match inputs to ONNX spec. For example, `Foo(x, y, nullptr, z)` will look the same as `Foo(x, y, z)`.

Reply:
ok, but for this line, how about `out << ",";`? I think there's no observable difference in the output

Reply:
Maybe let's use `""`? For example, `Foo("x", "y", "", "z")`.

Reply:
oh ok I see what you mean